instructor/common.py

```diff
@@ -0,0 +1,604 @@
+                                   'add_btn_num': add_btn_num,
+                                   'assignment': assignment
+                                  },
+                                  context_instance=RequestContext(request))
+
+
+def assignments_delete_strain(request):
+    models.Strains.objects.get(id=request.POST['pk']).delete()
+    return redirect('common_assignments_edit_strains')
+
+
+def assignments_edit_protocols(request):
+    pk = request.session['assignment_id']
+    assignment = models.Assignment.objects.get(id=pk)
+    message = ''
+    ProtocolsFormSet = modelformset_factory(models.Protocol, extra=1, can_delete=True, exclude=['assignment'])
+    if request.method == "POST":
+        formset = ProtocolsFormSet(request.POST)
+        formset.clean()
+        if ( formset.is_valid()):
+            message = "Thank you"
+            entries = formset.save(commit=False)
+            for form in entries:
+                form.assignment = assignment
+                form.save()
+        else:
+            message = "Something went wrong"
+
+    return render_to_response('instructor/protocols.html',
+                              {'formset': ProtocolsFormSet(
+                                  queryset=models.Protocol.objects.filter(assignment=assignment)),
+                               'message': message,
+                               'assignment': assignment
+                              },
+                              context_instance=RequestContext(request))
+
+
+def treatments_edit(request, assignment, protocol):
+    a = models.Assignment.objects.get(id=assignment)
+    p = models.Protocol.objects.get(id=protocol)
+    message = ''
+    treatments_set = ['treatment']
+    if a.has_concentration:
+        treatments_set.append('concentration')
+        treatments_set.append('concentration_unit')
+    if a.has_temperature:
+        treatments_set.append('temperature')
+    if a.has_start_time:
+        treatments_set.append('start_time')
+    if a.has_duration:
+        treatments_set.append('end_time')
+    if a.has_collection_time:
+        treatments_set.append('collection_time')
+
+    TreatmentsFormSet = modelformset_factory(models.Treatments, extra=1, can_delete=True, exclude=['protocol', 'order'],
+                                             can_order=True, fields=treatments_set)
+    if request.method == "POST":
+        formset = TreatmentsFormSet(request.POST)
+        formset.clean()
+        if ( formset.is_valid()):
+            message = "Thank you"
+            for form in formset.ordered_forms:
+                form.instance.order = form.cleaned_data['ORDER']
+            entries = formset.save(commit=False)
+            for form in entries:
+                form.protocol = p
+                form.save()
+        else:
+            message = "Something went wrong"
+
+    return render_to_response('instructor/treatments.html',
+                              {'formset': TreatmentsFormSet(
```

Comment:
this is a bit busy for assigning to context, do `formset = TreatmentsFormSet....` before the return and just have `'formset': formset` here